examples/03_datetime_encoder.py

```diff
@@ -35,72 +35,63 @@
 .. |HGBR| replace::
     :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
 
-.. |to_datetime| replace::
-    :func:`~skrub.to_datetime`
+.. |ToDatetime| replace::
+    :class:`~skrub.ToDatetime`
 """
 
 
 ###############################################################################
 # A problem with relevant datetime features
 # -----------------------------------------
 #
-# We will use a dataset of air quality measurements in different cities.
-# In this setting, we want to predict the NO2 air concentration, based
-# on the location, date and time of measurement.
+# We will use a dataset of bike sharing demand in 2011 and 2012.
+# In this setting, we want to predict the number of bike rentals, based
+# on the date, time and weather conditions.
 
 from pprint import pprint
 
 import pandas as pd
 
 data = pd.read_csv(
-    "https://raw.githubusercontent.com/pandas-dev/pandas"
-    "/main/doc/data/air_quality_no2_long.csv"
-).sort_values("date.utc")
+    "https://raw.githubusercontent.com/skrub-data/datasets/master"
+    "/data/bike-sharing-dataset.csv"
+)
 # Extract our input data (X) and the target column (y)
-y = data["value"]
-X = data[["city", "date.utc"]]
+y = data["cnt"]
+X = data[["date", "holiday", "temp", "hum", "windspeed", "weathersit"]]
 
 X
 
 ###############################################################################
-# We convert the dataframe date columns using |to_datetime|. Notice how
-# we don't need to specify the columns to convert.
-from skrub import to_datetime
+# We convert the dataframe ``"date"`` column using |ToDatetime|.
+
+from skrub import ToDatetime
+
+X["date"] = ToDatetime().fit_transform(X["date"])
```

Comment:
Here I would prefer not to modify X. we want to avoid modifying the input data in place and do transformations inside of the processing pipeline. and it isn't needed because the tablevectorizer will do it for us.

we can still apply ToDatetime if we want to show users that it exists, but then we can discard the result instead of assigning it to a column in X